official/v7/entity_aggregation.go

```diff
@@ -22,3 +22,40 @@ type RequestBucketAggregation struct {
 	Field         string
 	MetricsSubAgg []RequestMetricAggregation
 }
+
+// ResponseAggregation contains all the aggregations returned from elastic search
+type ResponseAggregation struct {
+	MetricAggregations []ResponseMetricAggregation
+	BucketAggregations []ResponseBucketAggregation
+}
+
+// AppendAggregation adds a parsed aggregation to the appropriate slice
+// in the ResponseAggregation. It handles ResponseMetricAggregation and
+// ResponseBucketAggregation types. If the input 'aggs' is of any other type,
+// it is ignored.
+func (a ResponseAggregation) AppendAggregation(
+	aggs interface{},
+) ResponseAggregation {
+	if metricAggs, ok := aggs.(ResponseMetricAggregation); ok {
+		a.MetricAggregations = append(a.MetricAggregations, metricAggs)
+	} else if bucketAggs, ok := aggs.(ResponseBucketAggregation); ok {
+		a.BucketAggregations = append(a.BucketAggregations, bucketAggs)
```

Comment:
Are metric aggregations and bucket aggregations always mutually exclusive?

Reply:
Yes. An aggregation will either be a metric aggregation or a bucket aggregation.